iosApp/iosApp/engine/IosOonimkallBridge.swift

```diff
@@ -1,23 +1,223 @@
 import composeApp
 import Oonimkall
 
-class IosOonimkallBridge : OonimkallBridge {
+let CONTEXT_TIMEOUT: Int64 = -1
+
+class IosOonimkallBridge: OonimkallBridge {
+
     func startTask(settingsSerialized: String) throws -> OonimkallBridgeTask {
         var error: NSError?
         let task = OonimkallStartTask(settingsSerialized, &error)!
 
-        class Task : OonimkallBridgeTask {
+        class Task: OonimkallBridgeTask {
             var task: OonimkallTask
-            init(task: OonimkallTask) { self.task = task }
-            func isDone() -> Bool { task.isDone() }
-            func interrupt() { task.interrupt() }
-            func waitForNextEvent() -> String { task.waitForNextEvent() }
+
+            init(task: OonimkallTask) {
+                self.task = task
+            }
+
+            func isDone() -> Bool {
+                task.isDone()
+            }
+
+            func interrupt() {
+                task.interrupt()
+            }
+
+            func waitForNextEvent() -> String {
+                task.waitForNextEvent()
+            }
         }
-    
+
         return Task(task: task)
     }
-    
+
     func doNewSession(sessionConfig: OonimkallBridgeSessionConfig) throws -> OonimkallBridgeSession {
-        fatalError("Not Implemented")
+        class IosSession: OonimkallBridgeSession {
+            private let sessionConfig: OonimkallSessionConfig
+
+            init(sessionConfig: OonimkallSessionConfig) {
+                self.sessionConfig = sessionConfig
+            }
+
+            func error(_ message: String, code: Int = 0, domain: String = "IosOonimkallBridge", function: String = #function, file: String = #file, line: Int = #line) -> NSError {
+
+                let functionKey = "\(domain).function"
+                let fileKey = "\(domain).file"
+                let lineKey = "\(domain).line"
+
+                let error = NSError(domain: domain, code: code, userInfo: [
+                    message: message,
+                    functionKey: function,
+                    fileKey: file,
+                    lineKey: line
+                ])
+
+                return error
+            }
```

Comment:
Ah, this is much more thorough than what I thought, which was just adding a description. But it looks great.